portfolio/src/main/java/com/google/sps/servlets/UpvoteCommentServlet.java

```diff
@@ -0,0 +1,61 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import java.util.List;
+import java.util.ArrayList;
+import com.google.gson.Gson;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+
+import com.google.sps.data.Comment;
+
+/** Servlet responsible for deleting comments. */
+@WebServlet("/upvote-comment")
+public class UpvoteCommentServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    long id = Long.parseLong(request.getParameter("id"));
+    long upvotes = Long.parseLong(request.getParameter("upvotes"));
```

Comment:
Does this solve the issue we talked about? If B upvotes before A, aren't you still overriding B's upvote when you eventually doPost for A?

Reply:
Probably not... The easiest fix for this I think is to rewrite the upvote method as a set as discussed. Looks like java's hashSet size and add feature are both constant time, so will only be more expensive space wise... but probably a fair tradeoff as opposed to figuring out how to do this atomically. 

Will add once Authentication is merged with master branch.

Reply:
The solution we talked about was reading the current upvote count from the data storage and incrementing that. That would fix the issue and does not need auth to work. Right now, this doesn't work as expected.

I'm gonna resolve because this is previous week's code, but you should try to fix this some time.